### PR TITLE
Reducing `to_string`s

### PR DESCRIPTION
**Pull request by robertbastian**
#2711

### CODE REVIEW

**Review by Manishearth — COMMENTED** [review]
I don't understand why we need to do this? It feels less readable overall, and ToString for str just calls String::from anyway

**Comment by robertbastian** [comment]
#2711

File: components/datetime/src/provider/calendar/symbols.rs [hunk 1]
```diff
@@ -351,13 +351,13 @@ mod test {
...
 
         let weekdays = weekdays::SymbolsV1([
-            Cow::Owned("Monday".to_string()),
-            Cow::Owned("Tuesday".to_string()),
-            Cow::Owned("Wednesday".to_string()),
-            Cow::Owned("Thursday".to_string()),
-            Cow::Owned("Friday".to_string()),
-            Cow::Owned("Saturday".to_string()),
-            Cow::Owned("Sunday".to_string()),
+            Cow::Owned(String::from("Monday")),
```
[the remark this answers is not in the document]
Comment (robertbastian) [reply components/datetime/src/provider/calendar/symbols.rs]: of course

**Comment by sffc** [comment]
Takeaways from discussion with ICU4X-SC:

Style guide for examples:

1. Compare fields instead of strings where possible (call parse instead of to_string)
2. Don't do anti-patterns in docs. Calling `to_string` on the output of a `format` function is an anti-pattern. Use `format_to_string` instead
3. Make docs readable with minimal context.
4. Write down tips for performance optimization somewhere, either in a tutorial, docs, etc.

**Comment by robertbastian** [comment]
Breaking locid and FixedDecimal changes into a different PRs.

**Comment by robertbastian** [comment]
@zbraniecki for style guide phrasing

**Review by Manishearth — APPROVED** [review]

**Review by sffc — APPROVED** [review]

**Comment by robertbastian** [comment]
Merging without Zibi's review as I'm going OOO

**Pull request merged**